src/com/amazon/ion/Timestamp.java

```diff
@@ -1374,14 +1387,48 @@ public static Timestamp forMillis(long millis, Integer localOffset)
      *
      * @throws NullPointerException if {@code millis} is {@code null}
      *
-
      */
     public static Timestamp forMillis(BigDecimal millis, Integer localOffset)
     {
         return new Timestamp(millis, localOffset);
     }
 
 
+    /**
+     * Returns a Timestamp that represents the point in time that is {@code seconds} from the unix epoch
+     * (1970-01-01T00:00:00.000Z), with the {@code nanoOffset} applied and a given local offset.
+     *
+     * <p>This function is intended to allow easy conversion to Timestamp from Java 8's {@code java.time.Instant}
+     * without having to port this library to Java 8.  The following snippet will yield a Timestamp {@code ts}
+     * that equivalent to the {@code java.time.Instant} {@code i}:</p>
+     *
+     * <code>
+     *     Instant i = Instant.now();
+     *     Timestamp ts = Timestamp.forEpochSecond(i.getEpochSecond(), i.getNano(), 0);
+     * </code>
+     *
+     * <p>Like {@link #forMillis}, {@code seconds} is relative to UTC, regardless of the value
+     * supplied for {@code localOffset}.</p>
+     *
+     * @param seconds The number of seconds from the unix epoch (1970-01-01T00:00:00.000Z) UTC.
+     * @param nanoOffset The number of nanoseconds for the fractional component. Must be between 0 and 999,999,999.
+     * @param localOffset the local offset from UTC, measured in minutes;
+     *                    may be {@code null} to represent an unknown local offset.
+     */
+    public static Timestamp forEpochSecond(long seconds, int nanoOffset, Integer localOffset) {
+        // We do not validate seconds here because that is done within the forMillis static constructor.
+        long millis = seconds * 1000L;
+        Timestamp ts = forMillis(millis, localOffset);
+        if(nanoOffset != 0) {
+            if(nanoOffset < 0 || nanoOffset > 999999999) {
+                throw new IllegalArgumentException("nanoOffset must be between 0 and 999,999,999");
+            }
+            ts._fraction = ts._fraction.add(BigDecimal.valueOf(nanoOffset).movePointLeft(9)).stripTrailingZeros();
```

Comment:
I'm not sure that stripping trailing zeros is the right thing to do here.

Since this method accepts a number of nanoseconds, I don't think it would be unreasonable for all Timestamps produced by this method to have nanosecond precision. If users want a different level of fractional precision, they can use one of the other factory methods to achieve that.

The behavior of this method with regard to fractional precision should definitely be documented either way.